sklearn/tests/test_common.py

```diff
@@ -105,7 +107,7 @@ def _tested_estimators():
         yield estimator
 
 
-@parametrize_with_checks(_tested_estimators())
+@parametrize_with_checks(list(_tested_estimators()))
```

Comment:
without this, the iterator would be consumed in `parametrize_with_checks` due to the addition of `if any(isinstance(est, type) for est in estimators):`

Note that the docstring of `parametrize_with_checks` says that its argument should be a list.